dspace/config/crosswalks/oai/metadataFormats/qdc.xsl

```diff
@@ -26,6 +26,16 @@
 			<xsl:for-each select="doc:metadata/doc:element[@name='dc']/doc:element[@name='title']/doc:element/doc:field[@name='value']">
 				<dc:title><xsl:value-of select="." /></dc:title>
 			</xsl:for-each>
+			<xsl:for-each select="doc:metadata/doc:element[@name='dc']/doc:element[@name='creator']/doc:element/doc:field[@name='value']">
+				<dc:creator>
+					<xsl:value-of select="." />
+				</dc:creator>
+			</xsl:for-each>
+			<xsl:for-each select="doc:metadata/doc:element[@name='dc']/doc:element[@name='contributor']/doc:element[@name='author']/doc:element/doc:field[@name='value']">
```

Comment:
Hi.

I made this patch.

It doesn’t exactly add a dc.contributor.author (there is no such thing in QDC).

The issue is only that there is no dc.creator element in the output.   The fact that it pulls data from the source document’s dc.creator and from the source document’s dc.contributor.author in order to populate the output document’s dc.creator doesn’t make it a separate issue, I don’t think.

Brad


From: Hardy Pottinger [mailto:notifications@github.com]
Sent: October 7, 2015 2:30 PM
To: DSpace/DSpace <DSpace@noreply.github.com>
Cc: Brad Dewar <brad.dewar@novanet.ns.ca>
Subject: Re: [DSpace] [DS-2792] Add dc.creator to the output of qdc.xsl (#1074)


In dspace/config/crosswalks/oai/metadataFormats/qdc.xsl<https://github.com/DSpace/DSpace/pull/1074#discussion_r41419762>:

> @@ -26,6 +26,16 @@

>                      <xsl:for-each select="doc:metadata/doc:element[@name='dc']/doc:element[@name='title']/doc:element/doc:field[@name='value']">

>                              <dc:title><xsl:value-of select="." /></dc:title>

>                      </xsl:for-each>

> +                    <xsl:for-each select="doc:metadata/doc:element[@name='dc']/doc:element[@name='creator']/doc:element/doc:field[@name='value']">

> +                            <dc:creator>

> +                                   <xsl:value-of select="." />

> +                            </dc:creator>

> +                    </xsl:for-each>

> +                    <xsl:for-each select="doc:metadata/doc:element[@name='dc']/doc:element[@name='contributor']/doc:element[@name='author']/doc:element/doc:field[@name='value']">

This change does more than add dc.creator, it also adds dc.contributor.author... which is probably a necessary change/bugfix, but has not yet been mentioned as an issue.

—
Reply to this email directly or view it on GitHub<https://github.com/DSpace/DSpace/pull/1074/files#r41419762>.